samcli/lib/utils/version_checker.py

```diff
@@ -0,0 +1,126 @@
+"""
+Contains information about newer version checker for SAM CLI
+"""
+import logging
+from datetime import datetime, timedelta
+
+import click
+from requests import get
+from samcli import __version__ as installed_version
+from samcli.cli.global_config import GlobalConfig
+
+LOG = logging.getLogger(__name__)
+
+AWS_SAM_CLI_PYPI_ENDPOINT = "https://pypi.org/pypi/aws-sam-cli/json"
+AWS_SAM_CLI_INSTALL_DOCS = (
+    "https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html"
+)
+PYPI_CALL_TIMEOUT_IN_SECONDS = 5
+DELTA_DAYS = 7
+
+
+def check_newer_version(func):
+    """
+    This function returns a wrapped function definition, which checks if there are newer version of SAM CLI available
+
+    Parameters
+    ----------
+    func: function reference
+        Actual function (command) which will be executed
+
+    Returns
+    -------
+    function reference:
+        A wrapped function reference which executes original function and checks newer version of SAM CLI
+    """
+
+    def wrapped(*args, **kwargs):
+        # execute actual command first
+        actual_result = func(*args, **kwargs)
+        # check and inform newer version if it is available
+        inform_newer_version()
+
+        return actual_result
+
+    return wrapped
+
+
+def inform_newer_version(check_all_always=False):
+    """
+    Compares installed SAM CLI version with the up to date version from PyPi,
+    and print information if up to date version is different then what is installed now
+
+    It will store last version check time into GlobalConfig, so that it won't be running all the time
+    Currently, it will be checking weekly
+
+    Parameters
+    ----------
+    check_all_always: bool
+        When it is True, it will trigger checking new version of SAM CLI. Default value is False
+
+    """
+    # run everything else in try-except block
+    global_config = None
+    need_to_update_last_check_time = True
+    try:
+        global_config = GlobalConfig()
+        last_version_check = global_config.last_version_check
+
+        if check_all_always or is_last_check_older_then_week(last_version_check):
+            compare_current_version()
+        else:
+            need_to_update_last_check_time = False
+    except Exception as e:
+        LOG.debug("New version check failed", exc_info=e)
+    finally:
+        if need_to_update_last_check_time:
+            update_last_check_time(global_config)
+
+
+def compare_current_version():
```

Comment:
I would rename this to `get_and_compare_versions`

Reply:
I would agree with this one, if we were returning something from this command, but it only outputs the difference.
Updating it as `fetch_and_compare_versions` as we discussed offline